pkg/packagekit/updates.jsx

```diff
@@ -457,56 +484,55 @@ class OsUpdates extends React.Component {
             .fail(ex => this.handleLoadError((ex.problem == "not-found") ? _("PackageKit is not installed") : ex));
     }
 
-    watchUpdates(transProxy) {
-        this.setState({state: "applying", applyTransaction: transProxy, allowCancel: transProxy.AllowCancel});
+    watchUpdates(transObj) {
+        this.setState({state: "applying", applyTransaction: transObj, allowCancel: false});
 
-        transProxy.addEventListener("ErrorCode", (event, code, details) => this.state.errorMessages.push(details));
+        dbus_pk.call(transObj, "DBus.Properties", "Get", [I_TRANS, "AllowCancel"])
+            .done(reply => this.setState({allowCancel: reply[0].v}));
 
-        transProxy.addEventListener("Finished", (event, exit) => {
-            this.setState({applyTransaction: null, allowCancel: null});
+        pkWatchTransaction(transObj,
+            {
+                ErrorCode: (code, details) => this.state.errorMessages.push(details),
 
-            if (exit == PK_EXIT_ENUM_SUCCESS)
-                this.setState({state: "updateSuccess", haveSecurity: false, loadPercent: null});
-            else if (exit == PK_EXIT_ENUM_CANCELLED) {
-                this.setState({state: "loading", haveSecurity: false, loadPercent: null});
-                this.loadUpdates();
-            } else {
-                // normally we get FAILED here with ErrorCodes; handle unexpected errors to allow for some debugging
-                if (exit != PK_EXIT_ENUM_FAILED)
-                    this.state.errorMessages.push(cockpit.format(_("PackageKit reported error code $0"), exit));
-                this.setState({state: "updateError"});
-            }
-        });
+                Finished: exit => {
+                    this.setState({applyTransaction: null, allowCancel: null});
 
-        transProxy.addEventListener("changed", (event, data) => {
-            if ("AllowCancel" in data)
-                this.setState({allowCancel: data.AllowCancel});
-        });
+                    if (exit == PK_EXIT_ENUM_SUCCESS)
```

Comment:
If this is a number, wouldn't `===` be more suited to use as the comparison operator?